### ISSUE TITLE
[Bug]: Missing robustness in CalDAV code

### ISSUE THREAD

**Issue by backmes** [nextcloud/server#39576]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

We experienced the following behavior when creating new calendar events with thunderbird
- sometimes the calendar events were duplicated for the receiver
- sometimes the created calendar events didn't show up at all at the nextcloud database

The latter behavior seemed to occur mostly, when invited persons were not in our organization (i.e. not in the LDAP directory) 

The latter part caused inconsistent views on the calendar, apparently the event was created successfully in thunderbird, but invited persons never received an invitation. The calendar app of nextcloud also didn't show the new event.

The log files showed man entries with `TypeError` in line 217 of `${ROOT}/apps/dav/lib/CalDAV/Schedule/IMipPlugin.php`

We tracked down the issue to some major changes in the iMIP code from NC 25.0.8 to NC 25.0.9. This code does not include the commit `a35b960` as NC 26 and NC27 do.

Applying the commit fixed the issue for us.

### Steps to reproduce

1. Install NC 25.0.9
2. Use LDAP autentification method
3. Create calendar event using thunderbird, invite people not in LDAP directory

### Expected behavior

Events should be created successfully

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

25

### Operating system

RHEL/CentOS

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 22.1 to 22.2)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.verified.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "version": "25.0.9.2",
        "dbtype": "mysql",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "installed": true,
        "defaultapp": "files,dashboard",
        "default_locale": "de_DE",
        "default_phone_region": "DE",
        "skeletondirectory": "\/var\/lib\/nextcloud\/skeleton",
        "tempdirectory": "\/var\/lib\/nextcloud\/tmp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "pipe",
        "overwrite.cli.url": "https:\/\/cloud.verified.de",
        "htaccess.RewriteBase": "\/",
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "mysql.utf8mb4": true,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "upgrade.disable-web": true,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "auth.webauthn.enabled": false,
        "loglevel": 0,
        "trashbin_retention_obligation": "auto, 14",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtpstreamoptions": {
            "ssl": {
                "allow_self_signed": true,
                "verify_peer": false,
                "verify_peer_name": false
            }
        },
        "app_install_overwrite": [
            "terms_of_service"
        ],
        "mail_smtpsecure": "tls"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - admin_audit: 1.15.0
  - bruteforcesettings: 2.5.0
  - calendar: 4.4.3
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contacts: 5.3.2
  - dashboard: 7.5.0
  - dav: 1.24.0
  - deck: 1.8.5
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_antivirus: 5.2.1
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - groupfolders: 13.1.4
  - guests: 2.5.0
  - impersonate: 1.12.1
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notes: 4.8.1
  - notifications: 2.13.1
  - oauth2: 1.13.1
  - password_policy: 1.15.0
  - polls: 5.2.0
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.4
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - spreed: 15.0.7
  - support: 1.8.0
  - systemtags: 1.15.0
  - tasks: 0.15.0
  - terms_of_service: 2.2.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - twofactor_totp: 7.0.0
  - updatenotification: 1.15.0
  - user_ldap: 1.15.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - contactsinteraction: 1.0.0
  - encryption
  - files_external
  - photos: 1.0.0
  - sharebymail: 1.7.0
  - survey_client: 1.5.0
  - suspicious_login
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Additional info

Used Thunderbird version: 102.13.0

**Comment by kesselb** [comment]
Thanks for the thorough analysis :+1:

**Comment by joshtrichards** [comment]
Fixed in #39591 in NC25.
